engine/sound/src/decoders/decoder_wav.cpp

```diff
@@ -130,8 +130,16 @@ namespace dmSoundCodec
         bool fmt_found = false;
         bool data_found = false;
 
-        dmSound::Result res = dmSound::SoundDataRead(sound_data, 0, sizeof(header), &header);
-        if (res != dmSound::RESULT_OK) {
+        uint32_t nread;
+        dmSound::Result res = dmSound::SoundDataRead(sound_data, 0, sizeof(header), &header, &nread);
+        if (res < dmSound::RESULT_OK)
+        {
+            dmLogError("Failed to read riff header: %d", res);
+            return RESULT_INVALID_FORMAT;
+        }
```

Comment:
To make code easier, I promoted the RESULT_PARTIAL_DATA to a positive number (=1).